pylint/checkers/base.py

```diff
@@ -1882,6 +1902,28 @@ def _raise_name_warning(self, node, node_type, name, confidence):
         self.add_message("invalid-name", node=node, args=args, confidence=confidence)
         self.stats["badname_" + node_type] += 1
 
+    def _name_valid_due_to_whitelist(self, name: str) -> bool:
+        good_names_patterns = self.config.good_names_rgxs
+        matches = [
+            match
+            for match in (pattern.search(name) for pattern in good_names_patterns)
+            if match is not None
+        ]
+        return len(matches) > 0 or name in self.config.good_names
+
+    def _name_invalid_due_to_blacklist(self, node_type: str, name: str, node) -> bool:
+        bad_names_patterns = self.config.bad_names_rgxs
+        matches = [
```

Comment:
Ditto regarding compiling these in `open()`

Reply:
done